test/disjointCoarseMallocPool.cpp

```diff
@@ -282,8 +286,10 @@ TEST_P(CoarseWithMemoryStrategyTest, disjointCoarseMallocPool_basic) {
     ASSERT_EQ(GetStats(prov).alloc_size, init_buffer_size);
 
     umfPoolDestroy(pool);
-    umfMemoryProviderDestroy(coarse_memory_provider);
-    umfMemoryProviderDestroy(malloc_memory_provider);
+    // Both coarse_memory_provider and malloc_memory_provider
```

Comment:
You can add a check in the test below that umfPoolDestroy doesn't destroy the upstream memory provider (`OWN_PROVIDER` flag not set and the new parameter set to false by default).

Reply:
@PatKamin There is already such test: `disjointCoarseMallocPool_simple1` for example

Reply:
Yes, what I meant is to add a check after `umfDestroyPool`, like `GTEST_NE(provider, nullptr)` in the test you've mentioned.

Reply:
Such check gives us nothing, because destroying a provider or not does not change the provider's pointer saved in a variable: after calling `umfMemoryProviderDestroy(coarse_memory_provider);` the `coarse_memory_provider` is still `non-NULL`.

Reply:
Ok